docs/source/en/tasks/object_detection.mdx

```diff
@@ -0,0 +1,487 @@
+...     batch["pixel_values"] = encoding["pixel_values"]
+...     batch["pixel_mask"] = encoding["pixel_mask"]
+...     batch["labels"] = labels
+...     return batch
+```
+
+## Training the DERT model
+You have done most of the heavy lifting in the previous sections, so now you are ready to train your model!
+The images in this dataset are still quite large, even after resizing. This means that finetuning this model will
+require at least one GPU.
+
+```py
+>>> import torch
+
+>>> device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+```
+
+Training involves the following steps:
+1. Load the model with `DetrForObjectDetection.from_pretrained` using the same checkpoint as in the preprocessing.
+2. Define your training hyperparameters in `TrainingArguments`.
+3. Pass the training arguments to `Trainer` along with the model, dataset, image processor, and data collator.
+4. Call `train()`` to finetune your model.
+
+When loading the model from the same checkpoint that you used for the preprocessing, remember to pass the label to
+id and id to label maps that you created earlier from the dataset's metadata.
+
+```py
+>>> from transformers import DetrForObjectDetection
+
+>>> model = DetrForObjectDetection.from_pretrained(
+...     checkpoint,
+...     id2label=id2label,
+...     label2id=label2id,
+...     ignore_mismatched_sizes=True,
+... )
+```
+
+In the `TrainingArguments` use `output_dir` to specify where to save your model, then configure hyperparameters as you see fit.
+An important note: do not remove unused columns because this will drop the image column. Without the image column, you
+can't create `pixel_values`. For this reason, set `remove_unused_columns` to `False`.
+If you wish to share your model by pushing to the Hub, set `push_to_hub` to `True` (you must be signed in to Hugging
+Face to upload your model).
+
+```py
+>>> from transformers import TrainingArguments
+
+>>> training_args = TrainingArguments(
+...     output_dir="detr-resnet-50_fine_tuned_cppe5",
+...     per_device_train_batch_size=8,
+...     num_train_epochs=10,
+...     fp16=False,
+...     save_steps=200,
+...     logging_steps=50,
+...     learning_rate=1e-5,
+...     save_total_limit=2,
+...     remove_unused_columns=False,
+...     push_to_hub=True,
+... )
+```
+
+Finally, bring everything together, and call `train`:
+
+```py
+>>> from transformers import Trainer
+
+>>> trainer = Trainer(
+...     model=model,
+...     args=training_args,
+...     data_collator=collate_fn,
+...     train_dataset=cppe5["train"],
+...     tokenizer=image_processor,
+... )
+
+>>> rainer.train()
+```
+
+## Evaluate
+Object detection models are commonly evaluated with a set of [COCO-style metrics](https://cocodataset.org/#detection-eval).
+You can use one of the existing metrics implementations, here you'll use the one from `torchvision`.
+
+Save your trained model to use for evaluation. Alternatively, you can evaluate the model that you have pushed to the Hub.
```

Comment:
Note that we haven't yet called `trainer.push_to_hub()`, so I don't think the final model will be pushed to the Hub yet.

Reply:
Ooh, that's actually very subtle! I noticed that it was pushed to hub at checkpoints when we have `push_to_hub=True,` in the training arguments, and I assumed that it was pushing the final model too. If that's not the case, I definitely should add `trainer.push_to_hub()` after evaluation.

Reply:
Additionally, `trainer.push_to_hub()` will create the README for the model repository.

Reply:
Fixed in 1a38260b27ef977e0d82ba1c918eeb19c8f3dcb7